Obsidian/Commands/MainCommandModule.cs

```diff
@@ -331,6 +331,62 @@ public Task StopAsync(CommandContext ctx)
         return Task.CompletedTask;
     }
 
+    [Command("ban")]
+    [CommandInfo("Bans a user.", "/ban <user> <duration>")]
+    [RequirePermission(permissions: "obsidian.ban")]
+    public async Task BanAsync(CommandContext ctx, string username, int duration)
+    {
+        var server = (Server)ctx.Server;
+        var player = server.GetPlayer(username);
+        server.Bans.AddBan(player, duration);
+
+        if (ctx.IsPlayer)
+        {
+            await player.KickAsync($"{username} was banned by {ctx.Player.Username}");
+        }
+        else
+        {
+            await player.KickAsync($"{username} was banned by Console");
+        }
+    }
+
+    [Command("unban")]
+    [CommandInfo("Unbans a user.", "/unban <user>")]
+    [RequirePermission(permissions: "obsidian.ban")]
+    public async Task UnbanAsync(CommandContext ctx, string username)
+    {
+        var server = (Server)ctx.Server;
+        var player = server.GetPlayer(username);
+        server.Bans.RemoveBan(player);
+        
+        if (ctx.IsPlayer)
+        {
+            await player.KickAsync($"{username} was banned by {ctx.Player.Username}");
+        }
+        else
+        {
+            await player.KickAsync($"{username} was banned by Console");
+        }
+    }
+
+    [Command("kick")]
+    [CommandInfo("Kicks a user.", "/kick <user>")]
+    [RequirePermission(permissions: "obsidian.kick")]
+    public async Task KickAsync(CommandContext ctx, string username)
+    {
+        var server = (Server)ctx.Server;
+        var player = server.Players.FirstOrDefault(player => player.Username == username);
+        
+        if (ctx.IsPlayer)
+        {
+            await player.KickAsync($"{username} was banned by {ctx.Player.Username}");
+        }
+        else
+        {
+            await player.KickAsync($"{username} was banned by Console");
+        }
```

Comment:
Copy pasta fail, the disconnect message is wrong.

Reply:
Also, I suggest using ternary operator
```cs
await player.KickAsync($"{username} was banned by {(ctx.IsPlayer ? ctx.Player.Username : "Console")}");
```
Not sure if parentheses is required, check if it compiles without them